cylc/flow/scheduler.py

```diff
@@ -1102,17 +1099,20 @@ def remove_tasks(
 
         if flow_nums is None:
             flow_nums = set()
-        # Mapping of *relative* task IDs to removed flow numbers:
-        removed: Dict[Tokens, FlowNums] = {}
-        not_removed: Set[Tokens] = set()
+        # Mapping of relative task IDs to removed flow numbers:
+        removed: Dict[str, FlowNums] = {}
+        not_removed: Set[str] = set()
+        # All the matched tasks (will add applicable active tasks below):
+        matched_tasks = inactive.copy()
```

Comment:
Ok so I went back to using string IDs... But don't need the yucky `.split('/')` anymore